core/matrix/permutation.cpp

```diff
@@ -33,8 +33,219 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include <ginkgo/core/matrix/permutation.hpp>
 
 
+#include <ginkgo/core/base/exception_helpers.hpp>
+#include <ginkgo/core/base/executor.hpp>
+#include <ginkgo/core/base/precision_dispatch.hpp>
+#include <ginkgo/core/base/temporary_clone.hpp>
+#include <ginkgo/core/base/utils_helper.hpp>
+
+
+#include "core/base/dispatch_helper.hpp"
+#include "core/matrix/permutation_kernels.hpp"
+
+
 namespace gko {
 namespace matrix {
+namespace permutation {
+
+
+GKO_REGISTER_OPERATION(invert, permutation::invert);
+GKO_REGISTER_OPERATION(combine, permutation::combine);
+
+
+}  // namespace permutation
+
+
+template <typename IndexType>
+std::unique_ptr<const Permutation<IndexType>>
+Permutation<IndexType>::create_const(
+    std::shared_ptr<const Executor> exec, size_type size,
+    gko::detail::const_array_view<IndexType>&& perm_idxs,
+    mask_type enabled_permute)
+{
+    GKO_ASSERT_EQ(enabled_permute, row_permute);
```

Comment:
Is `enabled_permute` already in release? otherwise, we can omit this input

Reply:
This changes no interfaces, only behavior, with warnings for the cases where it's still used.